MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -1325,704 +1343,253 @@
-		disableAllMTLPixFmtCaps( BGRG422 );
-
-		disableMTLPixFmtCaps( RGB9E5Float, ColorAtt );
-
-		disableMTLPixFmtCaps( R8Unorm_sRGB, Write );
-		disableMTLPixFmtCaps( RG8Unorm_sRGB, Write );
-		disableMTLPixFmtCaps( RGBA8Unorm_sRGB, Write );
-		disableMTLPixFmtCaps( BGRA8Unorm_sRGB, Write );
-		disableMTLPixFmtCaps( PVRTC_RGBA_2BPP_sRGB, Write );
-		disableMTLPixFmtCaps( PVRTC_RGBA_4BPP_sRGB, Write );
-		disableMTLPixFmtCaps( ETC2_RGB8_sRGB, Write );
-		disableMTLPixFmtCaps( ETC2_RGB8A1_sRGB, Write );
-		disableMTLPixFmtCaps( EAC_RGBA8_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_4x4_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_5x4_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_5x5_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_6x5_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_6x6_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_8x5_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_8x6_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_8x8_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_10x5_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_10x6_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_10x8_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_10x10_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_12x10_sRGB, Write );
-		disableMTLPixFmtCaps( ASTC_12x12_sRGB, Write );
-	}
-#endif
-#endif
+void MVKPixelFormats::modifyMTLFormatCapabilities(const MVKMTLDeviceCapabilities& gpuCaps) {
+
+	bool noVulkanSupport =  false;		// Indicated supported in Metal but not Vulkan or SPIR-V.
+	bool notMac =  gpuCaps.isAppleGPU && !gpuCaps.supportsMac1;
+	bool iosOnly1 = notMac && !gpuCaps.supportsApple2;
+	bool iosOnly2 = notMac && !gpuCaps.supportsApple3;
+	bool iosOnly6 = notMac && !gpuCaps.supportsApple7;
+	bool iosOnly8 = notMac && !gpuCaps.supportsApple9;
+
+	setMTLPixFmtCapsIf( iosOnly2, A8Unorm, RF );
+	setMTLPixFmtCapsIf( iosOnly1, R8Unorm_sRGB, RFCMRB );
+	setMTLPixFmtCapsIf( iosOnly1, R8Snorm, RFWCMB );
+
+	setMTLPixFmtCapsIf( iosOnly1, RG8Unorm_sRGB, RFCMRB );
+	setMTLPixFmtCapsIf( iosOnly1, RG8Snorm, RFWCMB );
+
+	enableMTLPixFmtCapsIfGPU( Apple6, R32Uint, Atomic );
+	enableMTLPixFmtCapsIfGPU( Mac2,   R32Uint, Atomic );
+	enableMTLPixFmtCapsIfGPU( Apple6, R32Sint, Atomic );
+	enableMTLPixFmtCapsIfGPU( Mac2,   R32Sint, Atomic );
+
+	setMTLPixFmtCapsIf( iosOnly8, R32Float, RWCMB );
+
+	setMTLPixFmtCapsIf( iosOnly1, RGBA8Unorm_sRGB, RFCMRB );
+	setMTLPixFmtCapsIf( iosOnly1, RGBA8Snorm, RFWCMB );
+	setMTLPixFmtCapsIf( iosOnly1, BGRA8Unorm_sRGB, RFCMRB );
+
+	setMTLPixFmtCapsIf( iosOnly2, RGB10A2Unorm, RFCMRB );
+	setMTLPixFmtCapsIf( iosOnly2, RGB10A2Uint, RCM );
+	setMTLPixFmtCapsIf( iosOnly2, RG11B10Float, RFCMRB );
+	setMTLPixFmtCapsIf( iosOnly2, RGB9E5Float, RFCMRB );
+
+	// Blending is actually supported for RGB9E5Float, but format channels cannot
+	// be individually write-enabled during blending on macOS. Disabling blending
+	// on macOS is the least-intrusive way to handle this in a Vulkan-friendly way.
+	disableMTLPixFmtCapsIfGPU( Mac1, RGB9E5Float, Blend);
```

Comment:
Somebody had recently posted that this is now confirmed to be true in Vulkan as well. We should now be able to enable this.

Reply:
I think you're referring to #2302. 

It doesn't look like CTS has been modified for this yet. With latest CTS, we're still seeing several hundred blend tests failing.

We can track it in #2302 to support this once CTS is updated.

Reply:
Very well.